### PR TITLE
Adds getNewUsers() to registration flow

### PR DESCRIPTION
**Pull request by ottpeter**
Adds getNewUsers() to solve "registration needs re-entering the chat problem."
Some parameters were also adjusted.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "@ethersphere/bee-js": "^8.1.0",
     "@solarpunkltd/comment-system-ui": "^0.0.12",
     "@solarpunkltd/comment-system": "^1.1.5",
-    "@solarpunkltd/swarm-decentralized-chat": "^0.0.23",
+    "@solarpunkltd/swarm-decentralized-chat": "^0.1.0",
```
Comment (bosi95) [inline package.json:17]: please update package-lock.json too, build might fail

**Review by bosi95 — CHANGES_REQUESTED** [review]
please update package-lock.json too, build might fail

File: package.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
     "@ethersphere/bee-js": "^8.1.0",
     "@solarpunkltd/comment-system-ui": "^0.0.12",
```
Comment (bosi95) [inline package.json:15]: the latest version of comment-ui should be 0.0.13

**Review by bosi95 — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (bosi95) [reply package.json:15]: edit: it is updated to ^0.0.15 now

File: src/pages/Chat/Chat.tsx [hunk 3]
```diff
@@ -98,11 +95,44 @@ const Chat: React.FC<ChatProps> = ({
...
+        message: JSON.stringify({
+          text: messagesOlderThanThreshold[i].message,
+          threadId: messagesOlderThanThreshold[i].threadId,
+          messageId: messagesOlderThanThreshold[i].messageId,
+          parent: messagesOlderThanThreshold[i].parent
+        }),
+        timestamp: messagesOlderThanThreshold[i].timestamp
+      }
+      console.info("Resending message: ", newlyConstructedMessage)
+      const sResult = await chat.current?.sendMessage(ownAddress, topic, newlyConstructedMessage, stamp, privKey);
```
Comment (bosi95) [inline src/pages/Chat/Chat.tsx:123]: use a try-catch for sendMessage

File: src/components/ChatInput/ChatInput.tsx [hunk 4]
```diff
@@ -10,7 +10,6 @@ import SendIcon from "../icons/SendIcon/SendIcon";
 import { MessageWithThread, ThreadId } from "../../types/message";
 import { randomThreadId, handleKeyDown } from "../../utils/helpers";
 import InputLoading from "./InputLoading/InputLoading";
-import { BEING_SENT } from "../../utils/constants";
```
Comment (bosi95) [inline src/components/ChatInput/ChatInput.tsx:13]: remove being_sent from constants if not used at all

File: src/pages/Chat/Chat.tsx [hunk 5]
```diff
@@ -98,11 +95,44 @@ const Chat: React.FC<ChatProps> = ({
...
+      console.log("sResult ", sResult)
+    }
+  }
+
   useEffect(() => {
-    const messageIds = allMessages.map((msg) => JSON.parse(msg.message).messageId)
+    console.log("ALL MESSAGES: ", allMessages);
```
Comment (bosi95) [inline src/pages/Chat/Chat.tsx:129]: is is necessary to log every message (allMessages) ?

File: src/pages/Chat/Chat.tsx [hunk 6]
```diff
@@ -117,7 +147,7 @@ const Chat: React.FC<ChatProps> = ({
     for (let i = 0; i < data.length; i++) {
       let msgObj;
       try {
-        msgObj = JSON.parse(data[i].message);
+        msgObj = data[i].message as any;
```
Comment (bosi95) [inline src/pages/Chat/Chat.tsx:150]: why not parse ?

File: src/pages/HomePage/HomePage.tsx [hunk 7]
```diff
@@ -75,9 +49,9 @@ const HomePage: React.FC<HomePageProps> = ({ isLoaded, withGamification }) => {
 
   // We are reading user count values, when the component loads, and when selectedChat changes
   // (user is closing the Chat, and coming back to the Home Page)
-  useEffect(() => {
-    fetchUserCount();
-  }, [selectedChat]);
+//  useEffect(() => {
```
Comment (bosi95) [inline src/pages/HomePage/HomePage.tsx:52]: don't forget to remove this useEffect

File: src/pages/Chat/Chat.tsx (hunk 6 above)
Comment (ottpeter) [reply src/pages/Chat/Chat.tsx:150]: I think with AI filtering on the gateway proxy, something has changed, and parsing will break it.

File: src/pages/Chat/Chat.tsx (hunk 5 above)
Comment (ottpeter) [reply src/pages/Chat/Chat.tsx:129]: I removed it

File: src/components/ChatInput/ChatInput.tsx (hunk 4 above)
Comment (ottpeter) [reply src/components/ChatInput/ChatInput.tsx:13]: I removed it.

File: src/pages/Chat/Chat.tsx (hunk 3 above)
Comment (ottpeter) [reply src/pages/Chat/Chat.tsx:123]: Added try-catch.

**Pull request merged**